### PR TITLE
WRO-2575: Update moonstone samples with new React style APIs.

### PR DESCRIPTION
**Pull request by alexandrumorariu**
Enact-DCO-1.0-Signed-off-by: Alexandru Morariu alexandru.morariu@lgepartner.com
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Replaced all the class components with functional components from moonstone/sampler

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)

### Additional Considerations
[//]: # (How should the change be tested?)
[//]: # (Are there any outstanding questions?)
[//]: # (Were any side-effects caused by the change?)

### Links
[//]: # (Related issues, references)
WRO-2575

### Comments

### CODE REVIEW

File: samples/qa-a11y/src/views/EditableIntegerPicker.js [hunk 1]
```diff
@@ -2,38 +2,24 @@ import EditableIntegerPicker from '@enact/moonstone/EditableIntegerPicker';
...
-		super(props);
-		this.state = {
-			value: props.defaultValue
-		};
-	}
+	const text = 'The current length is ' + value + unit;
 
-	handleChange = (ev) => this.setState({value: ev.value});
+	return (
+		<EditableIntegerPicker {...props} aria-valuetext={text} onChange={handleChange} />
```
Comment (daniel-stoian-lgp) [inline samples/qa-a11y/src/views/EditableIntegerPicker.js]: The unit is not displayed for any value. Can you check?
image

Comment (alexandrumorariu) [reply samples/qa-a11y/src/views/EditableIntegerPicker.js]: Solved

File: samples/qa-a11y/src/views/ReadAlert.js [hunk 2]
```diff
@@ -21,56 +18,50 @@ class ReadAlertView extends Component {
...
+
+	const onClick = (clear) => () => readAlert('Enact is a framework designed to be performant, customizable and well structured.', clear);
 
-		this.onClick1 = this.onClick(true);
-		this.onClick2 = this.onClick(false);
-	}
 
-	onClick = (clear) => () => readAlert('Enact is a framework designed to be performant, customizable and well structured.', clear);
+	const onClick1 = useCallback(() => onClick(true), []);
+	const onClick2 = useCallback(() => onClick(false), []);
```
Comment (daniel-stoian-lgp) [inline samples/qa-a11y/src/views/ReadAlert.js]: The onClick method was not called with your implementation. here is a suggestion
```suggestion
	const onClick1 = onClick(true);
	const onClick2 = onClick(false);
```

File: samples/qa-videoplayer/src/views/MainPanel.js [hunk 3]
```diff
@@ -49,56 +35,53 @@ const SelectableVideoPlayer = class extends Component {
...
 		} else if (ev.time > selectionStart) {
 			// this.video.pause();
-			this.video.seek(selectionEnd);
+			videoRef.current.seek(selectionEnd);
 		}
 
 		// or remove the selection and allow the default behavior
 		// this.setState({
 		// 	selection: null
 		// });
```
Comment (daniel-stoian-lgp) [inline samples/qa-videoplayer/src/views/MainPanel.js]: can you remove this comment since it is obsolete after changing the component type?

**Review by daniel-stoian-lgp — APPROVED** [review]
LGTM. Awesome job!

File: samples/qa-videoplayer/src/views/MainPanel.js (hunk 3 above)
Comment (alexandrumorariu) [reply samples/qa-videoplayer/src/views/MainPanel.js]: Done

File: samples/qa-virtualgridlist/src/components/LocaleSwitch/LocaleSwitch.js [hunk 4]
```diff
@@ -1,6 +1,6 @@
 import {useI18nContext} from '@enact/i18n/I18nDecorator';
-import {useCallback} from 'react';
 import ToggleButton from '@enact/moonstone/ToggleButton';
+import {useCallback} from 'react';
 
 const LocaleSwitch = (props) => {
 	const {rtl, updateLocale} = useI18nContext();
```
Comment (adrian-cocoara-lgp) [inline samples/qa-virtualgridlist/src/components/LocaleSwitch/LocaleSwitch.js:6]: add an empty line

File: samples/qa-virtualgridlist/src/components/PanelHeader/PanelHeader.js [hunk 5]
```diff
@@ -1,13 +1,14 @@
+import kind from '@enact/core/kind';
 import Button from '@enact/moonstone/Button';
-import {connect} from 'react-redux';
 import Heading from '@enact/moonstone/Heading';
 import {Header} from '@enact/moonstone/Panels';
 import IconButton from '@enact/moonstone/IconButton';
 import Input from '@enact/moonstone/Input';
-import kind from '@enact/core/kind';
-import LocaleSwitch from '../LocaleSwitch';
 import PropTypes from 'prop-types';
```
Comment (adrian-cocoara-lgp) [inline samples/qa-virtualgridlist/src/components/PanelHeader/PanelHeader.js]: move import 'prop-types' below. before {connect} import

File: samples/qa-virtualgridlistnative/src/components/LocaleSwitch/LocaleSwitch.js [hunk 6]
```diff
@@ -1,6 +1,6 @@
 import {useI18nContext} from '@enact/i18n/I18nDecorator';
-import {useCallback} from 'react';
 import ToggleButton from '@enact/moonstone/ToggleButton';
+import {useCallback} from 'react';
 
 const LocaleSwitch = (props) => {
 	const {rtl, updateLocale} = useI18nContext();
```
Comment (adrian-cocoara-lgp) [inline samples/qa-virtualgridlistnative/src/components/LocaleSwitch/LocaleSwitch.js:6]: add an empty line

File: samples/qa-virtuallistnative/src/components/LocaleSwitch/LocaleSwitch.js [hunk 7]
```diff
@@ -1,6 +1,6 @@
 import {useI18nContext} from '@enact/i18n/I18nDecorator';
-import {useCallback} from 'react';
 import ToggleButton from '@enact/moonstone/ToggleButton';
+import {useCallback} from 'react';
 
 const LocaleSwitch = (props) => {
 	const {rtl, updateLocale} = useI18nContext();
```
Comment (adrian-cocoara-lgp) [inline samples/qa-virtuallistnative/src/components/LocaleSwitch/LocaleSwitch.js:6]: add an empty line

File: samples/qa-virtualgridlist/src/components/PanelHeader/PanelHeader.js (hunk 5 above)
Comment (alexandrumorariu) [reply samples/qa-virtualgridlist/src/components/PanelHeader/PanelHeader.js]: Done

File: samples/qa-virtualgridlistnative/src/components/LocaleSwitch/LocaleSwitch.js (hunk 6 above)
Comment (alexandrumorariu) [reply samples/qa-virtualgridlistnative/src/components/LocaleSwitch/LocaleSwitch.js:6]: Done

File: samples/qa-virtuallistnative/src/components/LocaleSwitch/LocaleSwitch.js (hunk 7 above)
Comment (alexandrumorariu) [reply samples/qa-virtuallistnative/src/components/LocaleSwitch/LocaleSwitch.js:6]: Done

File: samples/qa-virtualgridlist/src/components/LocaleSwitch/LocaleSwitch.js (hunk 4 above)
Comment (alexandrumorariu) [reply samples/qa-virtualgridlist/src/components/LocaleSwitch/LocaleSwitch.js:6]: Done

**Review by adrian-cocoara-lgp — APPROVED** [review]
Very nice job! LGTM

File: samples/qa-videoplayer/src/views/MainPanel.js [hunk 8]
```diff
@@ -49,56 +35,48 @@ const SelectableVideoPlayer = class extends Component {
...
-		this.video = video;
-	};
+	return (
+		<VideoPlayer
+			{...props}
+			loop
+			onSeekOutsideSelection={handleSeekOutsideSelection}
+			onTimeUpdate={handleTimeUpdate}
+			selection={selection}
+			ref={setVideo}
```
Comment (vladut-clapou-lgp) [inline samples/qa-videoplayer/src/views/MainPanel.js:67]: Can you sort the props please?

File: samples/qa-videoplayer/src/views/MainPanel.js [hunk 9]
```diff
@@ -49,56 +35,48 @@ const SelectableVideoPlayer = class extends Component {
...
+		>
+			<MediaControls>
+				<IconButton className={selecting ? css.selecting : ''} slot="rightComponents" onTap={handleToggleSelection}>repeat</IconButton>
```
Comment (vladut-clapou-lgp) [inline samples/qa-videoplayer/src/views/MainPanel.js]: Here as well, sort props please

File: samples/qa-virtualgridlist/src/views/MainView.js [hunk 10]
```diff
@@ -1,55 +1,49 @@
...
+	}, []);
+
+	const getScrollTo = useCallback((newScrollTo) => {
+		scrollTo.current = newScrollTo;
+	}, []);
+
+	return (
+		<div className={css.mainView}>
+			<PanelHeader
+				title="VirtualGridList"
```
Comment (vladut-clapou-lgp) [inline samples/qa-virtualgridlist/src/views/MainView.js]: Also here, sort props please

File: samples/qa-virtualgridlistnative-preserve-focus/src/components/SampleVirtualGridListNative.js [hunk 11]
```diff
@@ -1,47 +1,44 @@
...
+
+	const renderItem = useCallback(({index: receivedIndex, ...props}) => (
+		<Item {...props} onClick={onClick}>
+			{items[receivedIndex]}
 		</Item>
+	), [onClick]);
+
+	return (
+		<VirtualGridListNative
+			{...rest}
```
Comment (vladut-clapou-lgp) [inline samples/qa-virtualgridlistnative-preserve-focus/src/components/SampleVirtualGridListNative.js:25]: Also here, sort props please

File: samples/qa-virtuallist/src/views/MainPanel.js [hunk 12]
```diff
@@ -1,116 +1,93 @@
...
+					DataSize:
+					<Input
+						onChange={handleChange}
+						placeholder={`${listItems.length}`}
+						size="small"
+						style={{width: '5em'}}
+						type="number"
+						value={value}
+					/>
+					<Button size="small" onClick={onChangeDataSize}>Set DataSize</Button>
```
Comment (vladut-clapou-lgp) [inline samples/qa-virtuallist/src/views/MainPanel.js]: Also here, sort props please and for the following lines 74 and 75

File: samples/qa-virtuallistnative/src/views/MainPanel.js [hunk 13]
```diff
@@ -1,116 +1,93 @@
...
+					DataSize:
+					<Input
+						onChange={handleChange}
+						placeholder={`${listItems.length}`}
+						size="small"
+						style={{width: '5em'}}
+						type="number"
+						value={value}
+					/>
+					<Button size="small" onClick={onChangeDataSize}>Set DataSize</Button>
```
Comment (vladut-clapou-lgp) [inline samples/qa-virtuallistnative/src/views/MainPanel.js]: Also here, sort props please and also for lines 74 and 75

File: samples/sampler/stories/qa/ContextualPopupDecorator.js [hunk 14]
```diff
@@ -51,74 +50,60 @@ const renderSliderPopup = () => (
...
-				open={this.state.open}
-				showCloseButton
-			/>
-		);
-	}
-}
+	return (
+		<ContextualButton
+			{...props}
+			onClose={handleOpenToggle}
```
Comment (vladut-clapou-lgp) [inline samples/sampler/stories/qa/ContextualPopupDecorator.js]: Also here, sort props please

File: samples/sampler/stories/qa/ContextualPopupDecorator.js [hunk 15]
```diff
@@ -131,22 +116,19 @@ class ContextualPopupWithArrowFunction extends Component {
...
-					onClick={this.handleOnClick}
-				/>
-			</div>
-		);
-	}
-}
+	return (
+		<div {...rest} style={{display: 'flex', justifyContent: 'flex-end'}}>
+			<ContextualPopup
+				ref={setRef}
```
Comment (vladut-clapou-lgp) [inline samples/sampler/stories/qa/ContextualPopupDecorator.js]: Also here, sort props please

File: samples/qa-videoplayer/src/views/MainPanel.js (hunk 8 above)
Comment (alexandrumorariu) [reply samples/qa-videoplayer/src/views/MainPanel.js:67]: Done

File: samples/qa-videoplayer/src/views/MainPanel.js (hunk 9 above)
Comment (alexandrumorariu) [reply samples/qa-videoplayer/src/views/MainPanel.js]: Done

File: samples/qa-virtualgridlist/src/views/MainView.js (hunk 10 above)
Comment (alexandrumorariu) [reply samples/qa-virtualgridlist/src/views/MainView.js]: Done

File: samples/qa-virtuallist/src/views/MainPanel.js (hunk 12 above)
Comment (alexandrumorariu) [reply samples/qa-virtuallist/src/views/MainPanel.js]: Done

File: samples/qa-virtuallistnative/src/views/MainPanel.js (hunk 13 above)
Comment (alexandrumorariu) [reply samples/qa-virtuallistnative/src/views/MainPanel.js]: Done

File: samples/sampler/stories/qa/ContextualPopupDecorator.js (hunk 14 above)
Comment (alexandrumorariu) [reply samples/sampler/stories/qa/ContextualPopupDecorator.js]: Done

File: samples/sampler/stories/qa/ContextualPopupDecorator.js (hunk 15 above)
Comment (alexandrumorariu) [reply samples/sampler/stories/qa/ContextualPopupDecorator.js]: Done

File: samples/qa-virtualgridlistnative-preserve-focus/src/components/SampleVirtualGridListNative.js (hunk 11 above)
Comment (alexandrumorariu) [reply samples/qa-virtualgridlistnative-preserve-focus/src/components/SampleVirtualGridListNative.js:25]: Done

**Pull request merged**